### PR TITLE
Ensure getaccountsasync doesn't need a network call

### PR DESCRIPTION
**Pull request by MarkZuber**
This is an in-progress development.  Opened a PR to collaborate.

### CODE REVIEW

File: msal/src/Microsoft.Identity.Client/TokenCache.cs [hunk 1]
```diff
@@ -754,13 +754,8 @@ private string GetPreferredEnvironmentHost(string environmentHost, InstanceDisco
             return preferredEnvironmentHost;
         }
 
-        internal async Task<IEnumerable<IAccount>> GetAccountsAsync(string authority, bool validateAuthority, RequestContext requestContext)
+        internal Task<IEnumerable<IAccount>> GetAccountsAsync(string authority, bool validateAuthority, RequestContext requestContext)
```
Comment (bgavrilMS) [inline msal/src/Microsoft.Identity.Client/TokenCache.cs]: If this is internal and non-async, does it still make sense to return a Task ?

**Review by bgavrilMS — APPROVED** [review]
LGTM.

File: msal/src/Microsoft.Identity.Client/TokenCache.cs (hunk 1 above)
Comment (bgavrilMS) [inline msal/src/Microsoft.Identity.Client/TokenCache.cs]: I think we should do the same modification for RemoveAccount, to be consistent?

CC @jmprieur @henrik-me

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 2]
```diff
@@ -209,15 +209,20 @@ internal TokenCache UserTokenCache
...
                 requestContext.Logger.Info("Token cache is null or empty. Returning empty list of accounts.");
-                return Enumerable.Empty<Account>();
             }
-            return await UserTokenCache.GetAccountsAsync(Authority, ValidateAuthority, requestContext).ConfigureAwait(false);
+            else
+            {
+                accounts = UserTokenCache.GetAccounts(Authority, ValidateAuthority, requestContext);
+            }
+
+            return Task.FromResult(accounts);
```
Comment (jennyf19) [inline msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs:225]: nice improvements here.

**Review by jennyf19 — APPROVED** [review]

**Comment by jennyf19** [comment]
This is a summary of the testing i've done w/the build of this branch:

**GetAccountsAsync() Testing w/b2clogin.com:**
B2C xamarin - Android, iOS, UWP
**GetAccountsAsync() Testing w/login.microsoftonline.com:**
B2C xamarin - UWP

**v1 to v2 sample w/corp account**
msalv.prbuild to adalv4 to adalv3
adalv3 to msalv.prbuild to adalv4
adalv4 to msalv.prbuild to adalv3

**B2C wpf app**
FB account
Able to access and delete account off-line

**Review by henrik-me — APPROVED** [review]
:shipit:

**Pull request merged**